### DISCUSSION TITLE
Optimize Search Button while driving

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#6] [Q&A] [answered]
First I have discovered a little bug. Normally when long pressing the Minus or Plus Button, the map zooms in stages what is very good. But this doesn't happen consequently. In some situations it doesn't work: long pressing on minus for zooming out does nothing. On the display it looks like this: only the minus and plus button is available, the others disappear. 

![grafik](https://user-images.githubusercontent.com/121371629/210202646-24f938f5-6952-404b-8d91-abdcc25273f8.png)

You first have to wait until the buttons disappear and try again, then it works again. 

Now the proposal for improvement: 
The search button has no significance while driving since no one searches for location while driving. You cannot insert any waypoint in the route anyway (a new wp can only be inserted when the route is stopped, what I by the way don't find so good; this was better in the past). This search button could be changed into a WOW button (also with other symbol), means: while driving this could be used to save the actual gps position as a reminder for places to remember. These could be saved as POIS in a separate WOW folder to be able to make a named favorite later on.

![grafik](https://user-images.githubusercontent.com/121371629/210203142-b16bd4b3-8490-437f-98b5-cabe12b105d1.png)

What do you think?

Maybe it would also be nice to make 2 WOW buttons, one on the left side and one on the right side with functions "WOW button for nice locations" and the other "WOW button for magnificent motorcycle roads" and save them into different folders "WOW locations" and "WOW roads".

**Comment by devemux86** [comment]
Thanks for the detailed report!

> In some situations it doesn't work: long pressing on minus for zooming out does nothing.

Maybe it was time for them to auto hide?
So they didn't get the long press action (I will check it).

> On the display it looks like this: only the minus and plus button is available, the others disappear.

The zoom buttons behave a bit independently of the others on auto hide, we probably need to bind them all together.

> You cannot insert any waypoint in the route anyway (a new wp can only be inserted when the route is stopped, what I by the way don't find so good; this was better in the past).

Thanks for mentioning this! I wanted to discuss this decision with the users again.

_We should probably revert it back to how it worked before:_ :slightly_smiling_face:
**You will be able to change the route again without having to stop navigation.**

> The search button has no significance while driving since no one searches for location while driving.

So if we allow again the route to change while the navigation is running,
should we still hide the search button in the navigation?

> while driving this could be used to save the actual gps position as a reminder for places to remember.

Thanks, this is a very interesting idea!

What you suggest is essentially extra (optional) buttons in navigation,
_so users can quickly save their favorite GPS locations?_

As this will happen while driving and we cannot write names, what name should these favorites have?
Automatic names with GPS coordinates?

> WOW button for nice locations
> WOW button for magnificent motorcycle roads

Favorites are locations, i.e. pairs of latitude,longitude.
With "magnificent motorcycle roads" you still mean to store locations in a different category?

**Comment by DolomitesTom** [comment]
> You will be able to change the route again without having to stop navigation.

perfect!


>should we still hide the search button in the navigation?

good question. I would say yes, but only because of burning into OLED screens... on the other hand, my screen is littered with burned-in buttons, so no problem any more... :-)
For direct access it would be better to not hide this new WOW button (at least as optional setting)... or why not use a button which is on the screen anyway, for example long press on distance bar

![grafik](https://user-images.githubusercontent.com/121371629/210225622-f93e515f-5f4f-423d-9d66-705be328cfa3.png)

> What you suggest is essentially extra (optional) buttons in navigation,
> so users can quickly save their favorite GPS locations?

yes, nice locations, discovered while driving

> Automatic names with GPS coordinates?

GPS coordinates and street names if possible... or if online access is available, maybe the "plus Code" included in the name.

> With "magnificent motorcycle roads" you still mean to store locations in a different category?

 yes, like before incl. street name (or plus Code additionally)... it is just to have this in a different category focusing certain nice locations on the one hand and beautiful roads on the other hand.

**Comment by WolfgangZ42** [comment]
I already have a WOW button - a screenshot app as overlay (SCREENSHOT CAPTURE). Use it less for "nice place" or "incredible road" but for incorrect information in OSM, which I can correct after the tour. With the screenshot time I can determine the exact position in the track based on the screenshot contents. An extra WOW button (or even two) I would want to hide. But maybe a tap on the field for the current speed is the solution (short/long for two different folders):
![2023-01-02 14 20 34](https://user-images.githubusercontent.com/121806238/210238572-3c79203b-754b-419c-a69a-4aeed7f2e9e5.png)

**Reply by Ciutnik** [reply]
> I already have a WOW button - a screenshot app as overlay (SCREENSHOT CAPTURE). Use it less for "nice place" or "incredible road" but for incorrect information in OSM, which I can correct after the tour.

I do exactly the same! Tasker takes a screenshot and at the same time a commemorative selfie is taken with the front camera 😎 Then I use an app (Picture Manager on Android) to link these images to a stored GPX – so I know exactly where they were taken.

**Reply by DolomitesTom** [reply]
Did you read the privacy policy of Tasker? They share a lot of sensible information with third parties. I don't want that at all and would not use such an app.

**Reply by Ciutnik** [reply]
> Did you read the privacy policy of Tasker?

Yes. I read there (https://tasker.joaoapps.com/privacy.html) that:

> Tasker does not share your data with ANY THIRD PARTY unless you, the user, specifically set up a profile or task in Tasker to do so.

Of course, this may not be true, but using Android (and not only!) I am still at risk of losing privacy. Besides, I can't imagine my life without Tasker 😉

**Sorry for OT!**

**Reply by DolomitesTom** [reply]
All good, everyone is free to use what he/she wants 🙂. There are also people who use Tiktok, Twitter, Facebook or similar stuff. For me this is waste of time. I like more driving a motorcycle.

So everyone sets different priorities. In any case, the great thing is that demux tries to accommodate the needs of dedicated users, as long as they make sense.

**Reply by devemux86** [reply]
> the great thing is that demux tries to accommodate the needs of dedicated users, as long as they make sense

Always! :+1:

**Comment by devemux86** [comment]
> You will be able to change the route again without having to stop navigation.

It will be available again in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
> You will be able to change the route again without having to stop navigation.

Available in [Cruiser 3.0.8](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
works, thanks a lot
